### ISSUE TITLE
Please add support for Thaumatology - Ritual Path Magic

### ISSUE THREAD

**Issue by Chaogomu** [richardwilkes/gcs#96]
Support for RPM is partially implemented under the Monster Hunters data sets, however, it's not quite correct for a full RPM based game.

Thaumatology - Ritual Path Magic adds several advantages, and tweaks to existing advantages. An example would be Magery. RPM Magery does not provide skill bonuses to Thaumatology or to spellcasting. An added advantage is Ritual Adept, which is in the Monster Hunters data set, but the breakdown options are not.

Skills are where the main issue with the current data set show up. Per Rules, Path skills are limited to the lower of the character's Thaumatology skill or 12 plus their Magery level. Characters without Magery 0 are at -5 to all Path skills. The Monster Hunters data set is correct in that the Path skills default to Thaumatology -6. Monster Hunters has a different name for one of the Path Skills.

RPM also adds some equipment in the form of Workspace Kits and Grimoires. These have a variable price based on TL. These Items are found in the Monster Hunters data set but are only available at TL-8 prices.

**Comment by bdrossow** [comment]
I'm having trouble following the...

> [richardwilkes](/richardwilkes) transferred this issue from richardwilkes/gcs_library [6 days ago](#event-2147266593)

Wouldn't it be possible to simply create the advantage in a personal library?  The New Advantages & Skills are very versatile.

**Comment by richardwilkes** [comment]
Yes, possibly. I got the impression changes were needed that could not be achieved with the current data model, necessitating changes in the GCS code base. I have never used Ritual Path Magic, so don’t have a good idea of what it entails.

**Comment by Chaogomu** [comment]
The tricky part is that RPM path skills are limited by two things.

They cannot exceed your skill level in thaumatology but are also limited in that they cannot exceed 12 plus your level in magery.

I've not found a way to do this with a new skill.

**Comment by richardwilkes** [comment]
That’s because it isn’t possible with the current code, hence my moving this issue to here seems correct.

**Comment by richardwilkes** [comment]
One of the many variants of magic was added to GCS a few releases ago: Ritual Magic Spells. Do they help with this request at all?

**Comment by Chaogomu** [comment]
Ok, I've just tested Version 4.19.

It's not quite rules compliant but is a vast improvement. The skill caps for path skills are not in place, like the IQ plus magery hard cap. The Thaumatology requirement is there, but without a cap.

Without those skill caps it's very easy to make errors with a character build, but as long as the GM knows the limits of the code (as you referenced above) then it's fairly easy to catch.

Maybe tweak the prerequisite section? Or not, It just seems kind of pointless to add for just a single optional magic system.

If you did tweak things maybe add a check mark next to the box "and whose level is at least" with "equal to" 

This would be useful for when you want to cap other types of skills. I cannot think of what ones would benefit but there might be some. It would add the option to cap path skills to Thaumatology,

Still an enhancement, but no rush at all. My games have been on pause since early March.

**Comment by evsan** [comment]
I was thinking about adding RPM Grimoire. 
Ritual Magic Spells are not exactly what needed. Neither the Advantages. The RPM Spell has energy cost and it's formula is `(SUM(effects)+Modifiers)xGreater Effects Modifier`, e.g. `Lesser Sense Mind (2) + Greater Sense Body (2) + Range 100 miles (8); Greater Effects: 1 (x3) = (2 + 2 + 6)x3`

I used RPM in my games and I know how to create the spells. I guess I could realise that feature if you think it's useful.  

@richardwilkes  don't you mind if I reach you directly in Discord so we could discuss the details of realisation.

**Comment by richardwilkes** [comment]
@Chaogomu Yes, updating prereqs to allow specifying a limit based on some other thing is likely the right way to do this.

@evsan Anyone is welcome to contact me on the apps or apps-dev channel in the GURPS Discord. I'm there quite often.

**Comment by ubergrog** [comment]
Classic Celtic Myth also has a skill cap rule for Druidic spells which are limited to no higher than the level of their governing Tree Lore skill (p95)

**Comment by SavarWallk** [comment]
one can script the prerequisite of this one now.
